File: C++/024. Swap Nodes in Pairs.cpp
/**
 * Definition for singly-linked list.
 * struct ListNode {
 *     int val;
 *     ListNode *next;
 *     ListNode(int x) : val(x), next(NULL) {}
 * };
 */
class Solution {
public:
    ListNode* swapPairs(ListNode* head) {
        ListNode* result;
        ListNode* temp;
        result->next=head;
        for(ListNode* it=result;it->next!=NULL;it=it->next->next)
        {
            if(it->next->next==NULL)
            {
                return result->next;
            }
            else
            {
                temp=it->next;
                it->next=temp->next;
                temp->next=it->next->next;
                it->next->next=temp;
            }
        }
        return result->next;
    }
};

File: python/024. Swap Nodes in Pairs.py
# Definition for singly-linked list.
# class ListNode:
#     def __init__(self, x):
#         self.val = x
#         self.next = None

class Solution:
    def swapPairs(self, head):
        """
        :type head: ListNode
        :rtype: ListNode
        """
        helper = node = ListNode(0)
        node.next = head
        while node.next and node.next.next:
            temp = node.next
            node.next = node.next.next
            temp.next = node.next.next
            node.next.next = temp
            node = node.next.next
        return helper.next

